sympy/printing/printer.py

```diff
@@ -287,15 +287,18 @@ def _print(self, expr, **kwargs):
                 printmethod = '_print_' + cls.__name__
                 if hasattr(self, printmethod):
                     return getattr(self, printmethod)(expr, **kwargs)
-            # Unknown object, fall back to the emptyPrinter. Checks what type of
-            # decimal separator to print.
-            if (self.emptyPrinter == str) & \
-                (self._settings.get('decimal_separator', None) == 'comma'):
-                expr = str(expr).replace('.', '{,}')
+            # Unknown object, fall back to the emptyPrinter.
             return self.emptyPrinter(expr)
         finally:
             self._print_level -= 1
 
+    def emptyPrinter(self, expr):
+        # Checks what type of decimal separator to print.
+        expr = str(expr)
```

Comment:
This overwrites the emptyPrinter defined above. Is there any reason this has to be done in the empty printer like this?

Reply:
The one defined as `= str`?

Reply:
Yes

Reply:
Split to #19333